src/server/server.cpp

```diff
@@ -65,18 +65,25 @@ int Server::main(int argc, char** argv) {
             std::cerr << "unknown server type: \"" << type << "\"\n";
         } else {
             server->readConfig(args);
-            server->setupSocket();
-            servers.push_back(server);
+            if (server->setupSocket() == 0) {
+                servers.push_back(server);
+            } else {
+                std::cerr << "Server '" << type << "' setup failed" << std::endl;
+            }
         }
     }
 
-    for (SocketServer* server: servers) {
-        server->start();
-    }
+    if (servers.size() == 0) {
+        std::cerr << "No valid servers configured." << std::endl;
```

Comment:
There's no logger in this code (yet?) but for now, I think it would b e a good idea to prefix this with `WARNING:`.